### PR TITLE
Remove hacks

### PR DESCRIPTION
**Pull request by zhouwein**

### CODE REVIEW

**Comment by sughandj** [comment]
I think we should stick to Linux based system for debugging because we never know if some future code would cause troubles. So, sys.path should remain as it is (It isn't hacky according to me). As for gitignore, we can avoid stream files without trouble.

**Comment by wangx173** [comment]
I think we should stick to Linux based system too, cuz it was developed specifically for Linux

**Comment by zhouwein** [comment]
>I think we should stick to Linux based system for debugging because we never know if some future code would cause troubles.

Right, but the offending code that breaks windows compatibility is line 24 of article_explorer.py, and that's actually not needed, even on a linux environment. I suspect it's only needed on mathlab machines and on other servers (no pip --user install), it doesn't point to a django installation.

>So, sys.path should remain as it is (It isn't hacky according to me)

Putting it in the shell script is slightly less hacky. Also, the environment variable that's supposed to be used is PYTHONPATH (only works for python modules), rather than PATH (for executable too).

tl;dr: doesn't break anything, slight improvement.

**Comment by Sunakujira1** [comment]
>line 24 of article_explorer.py, and that's actually not needed, even on a linux environment. I suspect it's only needed on mathlab machines

We have mathlab server, we are running on mathlab machine currently, and so we still need it.

> Also, the environment variable that's supposed to be used is PYTHONPATH (only works for python modules), rather than PATH (for executable too).

You're confused with python's sys.path and system's PATH variable:
sys.path - [A list of strings that specifies the search path for modules. Initialized from the environment variable PYTHONPATH, plus an installation-dependent default.](https://docs.python.org/2/library/sys.html#sys.path)
tl;dr: same feature, not any hacky-er, no improvement.

Keep in mind that the main motive for being against this is that developing on different environment have and will cause extra issues compared to developing on same (or similar) environment. This is even more crucial as testing on the real environment becomes more distant, and thus will slack doing it (have as well). It might be a hassle to adapt to developing/debugging on linux system, but considering the risk without it, I encourage you to start using linux system as main developing/debugging system.

**Comment by zhouwein** [comment]
why reopen it?
